### DISCUSSION TITLE
lc/973/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#2525] [Announcements]
# lc/973/

多种编程语言实现 LeetCode、《剑指 Offer（第 2 版）》、《程序员面试金典（第 6 版）》题解

https://leetcode.doocs.org/lc/973/

**Comment by hopeckel** [comment]
//PriorityQueue
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        if (points == null) {
            return null;
        }

        PriorityQueue<int[]> que = new PriorityQueue<>((a, b) ->
                (a[0] * a[0] + a[1] * a[1]) - (b[0] * b[0] + b[1] * b[1])
        );
        for (int[] point : points) {
            que.offer(new int[]{point[0], point[1]});
        }
        int[][] res = new int[k][2];
        for (int i = 0; i < k; i++) {
            res[i][0] = que.peek()[0];
            res[i][1] = que.poll()[1];
        }

        return res;
    }
}
